microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/MicroProfilePropertiesListenerManager.java

```diff
@@ -159,6 +163,12 @@ public boolean visit(IResourceDelta delta) throws CoreException {
 					event.setProjectURIs(new HashSet<String>());
 					event.getProjectURIs().add(JDTMicroProfileUtils.getProjectURI(file.getProject()));
 					fireAsyncEvent(event);
+				} else if (isConfigSource(file) && isFileContentChanged(delta)) {
+					MicroProfilePropertiesChangeEvent event = new MicroProfilePropertiesChangeEvent();

```

Comment:
Why you need that? It's for refreshing validation in Java file when application.properties changed? I find it's really shame to do that because you will compute all properties from Java sources when application.properties is saved although we will have none changed for properties.

I wonder if we should manage that in MP LSside when didSave is called, we publish diagnostics for all opened java files.

Reply:
I tried the strategy you mentioned (calling on `didSave`) but since the info is pulled from the target application.properties, it wouldn't actually update properly unless you saved twice (I guess once to trigger JDT to put the application.properties in target, one more for microprofile-jdt to update). The same thing was true for running on `didChange`, since the target is only updated on save. If I run on `didSave` and `didChange` it wasn't too bad, but the diagnostics were still noticeably out of sync of the content of application.properties.

Reply:
Ok that's maake sense, but perhaps we should manage an another type (ex : MicroProfilePropertiesScope.ONLY_PROPERTIES) to avoid evict sources properties on MP LS side an drecompute the sources properties. Let's keep like this and please create an issue for that.

Reply:
https://github.com/eclipse/lsp4mp/issues/162